File: src/main/java/org/zenbot/szolnok/timetable/batch/domain/BusRoute.java
package org.zenbot.szolnok.timetable.batch.domain;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@Document
public class BusRoute {

    @Id
    private String id;
    private String startBusStop;
    private String endBusStop;
    private List<BusStop> busStops;

    public void addBusStopTimetable(BusStop busStop) {
        busStops.add(busStop);
    }
}

File: szolnok-timetable-backend-domain/src/main/kotlin/org/zenbot/szolnok/timetable/backend/domain/document/bus/BusRoute.kt
package org.zenbot.szolnok.timetable.backend.domain.document.bus

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class BusRoute(
    @Id var id: String? = null,
    var startBusStop: String = "",
    var endBusStop: String = "",
    var busStops: MutableList<BusStop> = ArrayList()
) {

    fun addBusStopTimetable(busStop: BusStop) {
        busStops.add(busStop)
    }
}